File: chatuikit/src/main/java/com/cometchat/chatuikit/shared/interfaces/DateTimeFormatterCallback.java
package com.cometchat.chatuikit.shared.interfaces;

public interface DateTimeFormatterCallback {
    default String time(long timestamp) {return null;}

    default String today(long timestamp) {return null;}

    default String yesterday(long timestamp) {return null;}

    default String lastWeek(long timestamp) {return null;}

    default String otherDays(long timestamp) {return null;}

    default String minute(long timestamp) {return null;}

    default String minutes(long diffInMinutesFromNow, long timestamp) {return null;}

    default String hour(long timestamp) {return null;}

    default String hours(long diffInHourFromNow, long timestamp) {return null;}

}

File: chatuikit-kotlin/src/main/java/com/cometchat/uikit/kotlin/shared/interfaces/DateTimeFormatterCallback.kt
package com.cometchat.uikit.kotlin.shared.interfaces

interface DateTimeFormatterCallback {
    fun time(timestamp: Long): String? = null
    fun today(timestamp: Long): String? = null
    fun yesterday(timestamp: Long): String? = null
    fun lastWeek(timestamp: Long): String? = null
    fun otherDays(timestamp: Long): String? = null
    fun minute(timestamp: Long): String? = null
    fun minutes(diffInMinutesFromNow: Long, timestamp: Long): String? = null
    fun hour(timestamp: Long): String? = null
    fun hours(diffInHourFromNow: Long, timestamp: Long): String? = null
}